### PR TITLE
[Storage] Migrate `test_large_file.py` to test proxy

### PR DESCRIPTION
**Pull request by vincenttran-msft**
This PR migrates `test_large_file.py` and `test_large_file_async.py` to test proxy. 

TODO:
- [x] Run live test pipeline after CI passes (previously reported that live test pipeline was failing on Linux)

EDIT: Still failing in live on Linux Pypy. Adding back in skip
EDIT: Trying out a pypy3 specific skip (thanks to logic found in `conftest.py` for Datalake)

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/tests/test_large_file.py [hunk 1]
```diff
@@ -74,14 +75,17 @@ def test_append_large_stream_without_network(self, datalake_storage_account_name
...
         response = file_client.append_data(data, 0, LARGEST_BLOCK_SIZE)
 
-        self.assertIsNotNone(response)
-        self.assertEqual(self.payload_dropping_policy.append_counter, 1)
-        self.assertEqual(self.payload_dropping_policy.append_sizes[0], LARGEST_BLOCK_SIZE)
+        assert response is not None
+        assert self.payload_dropping_policy.append_counter == 1
+        assert self.payload_dropping_policy.append_sizes[0] == LARGEST_BLOCK_SIZE
 
     @pytest.mark.skip(reason="Pypy3 on Linux failed somehow, skip for now to investigate")
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-datalake/tests/test_large_file.py:82]: Can we make this Skip Pypy specific? I believe there is a way to do this with `@pytest.mark.skipif` and some form of version check. The version check may be similar to the PyPy check in the datalake conftest.

Comment (vincenttran-msft) [reply sdk/storage/azure-storage-file-datalake/tests/test_large_file.py:82]: Ooo. I will give that a try!

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**